base/linalg/dense.jl

```diff
@@ -18,6 +20,8 @@ function scale!{T<:BlasFloat}(X::Array{T}, s::T)
     X
 end
 
+scale!{T<:BlasFloat}(s::T, X::Array{T}) = scale!(X, s)
```

Comment:
I also thought it was weird. I guess it is there for non commutative numbers in the general case.